commit 5dca9eea1bd4263bfa4d037ab2443de1cd730f7e	5dca9eea1bd4263bfa4d037ab2443de1cd730f7e
Author: Daniel Veillard <veillard@redhat.com>
Date:   Fri Apr 7 17:13:28 2017 +0200

    Increase buffer space for port in HTTP redirect support
    
    For https://bugzilla.gnome.org/show_bug.cgi?id=780690
    
    nanohttp.c: the code wrongly assumed a short int port value.

diff --git a/nanohttp.c b/nanohttp.c
index e109ad75..373425de 100644
--- a/nanohttp.c
+++ b/nanohttp.c
@@ -1423,9 +1423,9 @@ retry:
     if (ctxt->port != 80) {
 	/* reserve space for ':xxxxx', incl. potential proxy */
 	if (proxy)
-	    blen += 12;
+	    blen += 17;
 	else
-	    blen += 6;
+	    blen += 11;
     }
     bp = (char*)xmlMallocAtomic(blen);
     if ( bp == NULL ) {
